### ISSUE TITLE
[Feature Request] Break long posts the same way it's done on desktop

### ISSUE THREAD

**Issue by floppyD** [K1rakishou/Kuroba#12]
Cheese Implementation is weird, it moves the whole post down if it's long enough instead of just breaking it.

**Comment by namazso** [comment]
or just letting us choose between the three:
- Breaking
- Pushing content lower
- Having empty space on the left

**Comment by Astridchan** [comment]
Some anon suggested using FlowTextView in the Kuroba thread:
https://boards.4channel.org/g/thread/71876522#p71893505
https://boards.4channel.org/g/thread/71876522#p71894368
https://boards.4channel.org/g/thread/71876522#p71894552

**Comment by Adamantcheese** [comment]
tl;dr: [FlowTextView](https://github.com/deano2390/FlowTextView) won't work as a drop-in replacement and would require a large amount of modification to make work with the current architecture. Maybe use "Overchan"'s implementation if you want proper flowing?

This is in relation to the original issue posted [here.](https://github.com/Floens/Clover/issues/585) Floens' comment still stands in that issue; text wrapping is hard. Partly due to technical issues (to be enumerated below) and partly due to design, which was explained in that issue quite well. I'll avoid reiterating what was said for the design part and focus on the technical issues of FlowTextView in relation to PostCells.

If you take a look at FlowTextView's example, it's a drop-in replacement for a RelativeLayout, with an extra field for text that will flow around child views. So if we do that for our PostCell implementation, we should just have it wrap around everything right? Set up the view, make it flow in the code once everything else is set up and we're good, right? Not really. In my testing it out for a text-flowed cell, it wrapped around the title and details sections fine, but it didn't like wrapping around the thumbnails. This is likely because the thumbnails weren't loaded yet, and were wrapping around something internal to the ThumbnailView itself or it was just ignoring it altogether. And then the actual text size was impressively incorrect as well, which would have needed to be fixed in code to account for the font size setting. But that wasn't the biggest issue, both of those are likely easily resolved. The biggest issue arises with PostLinkable spans inside the text itself. Those spans aren't actually displayed in the text due to it's limited support for spanned content (only spans from Html.fromHtml are accounted for). Which means that all of the following were broken:

1) Spoilered text
2) Reply links
3) Cross-board reply links
4) Board links
5) Board search links
6) [Probably regular links as well due to the AutoLink helper class also using PostLinkables as a wrapping span](https://github.com/Floens/Clover/blob/d42b9dfa069c4d157279d5d03ff7294bfe05c477/Clover/app/src/main/java/org/floens/chan/core/site/parser/CommentParserHelper.java)

And I don't know about you, but not having reply links kinda sorta completely makes the application useless for the most part. That's where I abandoned looking at flowing the text because it would cause major issues like that to arise if using a pre-made flow view. Another idea would be to use WebView instances for every cell, but that's gonna be hell on performance despite having access to all those nice CSS rules that would make this a piece of cake. So that's out. I decided to go with a hybrid approach then, only shift the text down if it was long enough to make it look bad. Originally I put this at "2 lines in the original configuration", however that was changed to be "the total height of the title, details, icons, and comment is all 1.5x taller than the thumbnail" before shifting the comment below; this value is up to personal preference. It works well enough in most situations and does eliminate the left hand blank space. However, as a result of doing all that, the actual recycler view itself isn't allowed to recycle PostCell instances, as their internal layout paramenters vary and you aren't really able to reset those paramenters easily due to how the recycler view functions, which can be a performance problem. In my testing it wasn't enough of an issue unless you were scrolling very fast.

Now [Overchan does flow its text](https://github.com/miku-nyan/Overchan-Android/blob/b4bd4dc437c432ce4eeb3081870989343a3f7b1d/src/nya/miku/wishmaster/ui/presentation/FlowTextHelperImpl.java), however it's a custom rolled text flow engine, not FlowTextView. It's possible that using that code would work, but I don't know what sort of assumptions that code is making about the view itself. I have good reason to believe that the implementation is taken from [this StackOverflow link](https://stackoverflow.com/questions/2248759/how-to-layout-text-to-flow-around-an-image), which may provide more context and be a better resource overall if it is implemented.

**Comment by namazso** [comment]
And how about just bringing back a switch for old / new style for the time? Having a relatively big phone on low dpi scaling makes the shifted-down text look worse than 10% lost screen space imo.
